abstract/group.go

```diff
@@ -49,6 +49,9 @@ type Scalar interface {
 
 	// Set to the modular inverse of scalar a
 	Inv(a Scalar) Scalar
+	
+	// Set to the gcd of a and its module
+	Gcd(a Scalar) Scalar
```

Comment:
I am not a fan of this method because it is not consistent with the other methods, as it only takes one argument whereas it should at least take two (`GCD` takes two numbers, `Add` takes two numbers, vs `Inv` takes one number). I see currently two options:
+ Change the signature to 
```go
Gcd(a Scalar, b Scalar) Scalar
```
+ Put a method the `nist/` package, that takes two `Scalar` and returns the GCD out of it.
```go
nist.GCD(a,b) Scalar
```
or
```go
abstract.GCD(a,b) Scalar
```
I think the first option is cleaner, but @bford may not like it, in that case the second option is fine to me. @ineiti what do you think ?

Now the other questions is whether to follow the golang `big.Int` [signature](https://golang.org/pkg/math/big/#Int.GCD), which can also take the `x` and `y` parameters. I think it is ok to let as is, but it depends on the application. Sounds like @lihiid you don't need these values so it make sense to let as-is for now.